tests/aws-cpp-sdk-dynamodb-unit-tests/endpointParamTest.cpp

```diff
@@ -0,0 +1,104 @@
+'"','a','s','s','i','g','n','"',':','"','a','r','r','a','y','V','a','l','u','e','"','}',']',',','"',
+'e','n','d','p','o','i','n','t','"',':','{','"','u','r','l','"',':','"','h','t','t','p','s',':','/',
+'/','e','x','a','m','p','l','e','.','c','o','m','/','{','a','r','r','a','y','V','a','l','u','e','}',
+'"','}',',','"','t','y','p','e','"',':','"','e','n','d','p','o','i','n','t','"','}',',','{','"','c',
+'o','n','d','i','t','i','o','n','s','"',':','[',']',',','"','d','o','c','u','m','e','n','t','a','t',
+'i','o','n','"',':','"','e','r','r','o','r',' ','f','a','l','l','t','h','r','o','u','g','h','"',',',
+'"','e','r','r','o','r','"',':','"','n','o',' ','a','r','r','a','y',' ','v','a','l','u','e','s',' ',
+'s','e','t','"',',','"','t','y','p','e','"',':','"','e','r','r','o','r','"','}',']','}','\0'
+}};
+
+const char* DynamoDBEndpointRules::GetRulesBlob()
+{
+    return RulesBlob.data();
+}
+
+
+class  DynamoDBEndpointProviderTest : public Endpoint::DynamoDBDefaultEpProviderBase
+{
+public:
+    using DynamoDBResolveEndpointOutcome = Aws::Endpoint::ResolveEndpointOutcome;
+
+    DynamoDBEndpointProviderTest()
+      : Endpoint::DynamoDBDefaultEpProviderBase(Aws::DynamoDB::DynamoDBEndpointRules::GetRulesBlob(), Aws::DynamoDB::DynamoDBEndpointRules::RulesBlobSize)
+    {}
+
+    ~DynamoDBEndpointProviderTest()
+    {
+    }
+
+};
+
+class EndpointTest : public Aws::Testing::AwsCppSdkGTestSuite {
+    
+
+    protected:
+    static const char ALLOCATION_TAG[];
+    
+   
+    public:
+    void SetUp() override{
+    }
+    
+};
+const char EndpointTest::ALLOCATION_TAG[] = "EndpointTest";
+
+
+
+TEST_F(EndpointTest, testStringArrayParam) {
+
+    std::shared_ptr<DynamoDBEndpointProviderTest> endpointProvider_sp = Aws::MakeShared<DynamoDBEndpointProviderTest>(ALLOCATION_TAG);
+
+    Aws::Endpoint::EndpointParameters parameters;
+    // Static context parameters
+    parameters.emplace_back(Aws::String("stringArrayParam"), Aws::Vector<Aws::String>{"staticValue2","staticValue1"}, Aws::Endpoint::EndpointParameter::ParameterOrigin::STATIC_CONTEXT);
+
+    auto res = endpointProvider_sp->ResolveEndpoint(parameters);
+    EXPECT_TRUE(res.IsSuccess());
+    //std::cout<<"url="<<res.GetResult().GetURL()<<std::endl;
+    EXPECT_EQ( res.GetResult().GetURL(), "https://example.com/staticValue2");
+   
+}
+
+
+TEST_F(EndpointTest, testStringArrayParamError) {
+
+    std::shared_ptr<DynamoDBEndpointProviderTest> endpointProvider_sp = Aws::MakeShared<DynamoDBEndpointProviderTest>(ALLOCATION_TAG);
+
+    Aws::Endpoint::EndpointParameters parameters;
+    // Static context parameters
+    parameters.emplace_back(Aws::String("stringArrayParam"), Aws::Vector<Aws::String>{}, Aws::Endpoint::EndpointParameter::ParameterOrigin::STATIC_CONTEXT);
+
+    auto res = endpointProvider_sp->ResolveEndpoint(parameters);
+    EXPECT_FALSE(res.IsSuccess());
+    //std::cout<<"url="<<res.GetResult().GetURL()<<std::endl;
```

Comment:
please do a run through and remove commented out code in the future

Reply:
ack